src/models/LeaderboardResults.ts

```diff
@@ -91,18 +89,52 @@ export class LeaderboardResults
   @UpdatedAt
   @Column(DataType.DATE)
   updatedAt!: Date;
+
+  static associations: {
+    _user: Association<LeaderboardResults, User>;
+    _leaderboardEntry: Association<LeaderboardResults, LeaderboardEntry>;
+    _leaderboardResultsMeta: Association<LeaderboardResults, LeaderboardResultsMeta>;
+  };
 }
 
-export function createLeaderBoardResult(
-  leaderBoardData: LeaderboardResultsBasicCreationAttributes,
+export function createOrUpdateLeaderBoardResult(
+  leaderBoardData: LeaderboardResultsCreationAttributes,
   transaction?: Transaction
 ) {
-  return LeaderboardResults.create(
+  return LeaderboardResults.upsert(
     {
       ...leaderBoardData,
-      createdAt: new Date(),
-      updatedAt: new Date(),
     },
-    { transaction }
-  );
+    {
+      transaction,
+    }
+  ).then(async (r) => {
+    if (r[0] && leaderBoardData._leaderboardResultsMeta) {
+      for (const meta of leaderBoardData._leaderboardResultsMeta) {
+        await LeaderboardResultsMeta.upsert(
+          {
+            ...meta,
+            _leaderboardResultsId: r[0].id,
+          },
+          { transaction }
+        );
+      }
+    }
+
+    return r;
+  });
```

Comment:
I have some mixed feelings on how I implemented this..
please share your thoughts if you have a more elegant approach